test/src/main/java/io/zeebe/test/exporter/ExporterIntegrationRule.java

```diff
@@ -0,0 +1,313 @@
+import io.zeebe.broker.system.configuration.BrokerCfg;
+import io.zeebe.broker.system.configuration.ExporterCfg;
+import io.zeebe.client.ClientProperties;
+import io.zeebe.client.api.subscription.JobHandler;
+import io.zeebe.client.api.subscription.JobWorker;
+import io.zeebe.exporter.record.Record;
+import io.zeebe.exporter.record.value.IncidentRecordValue;
+import io.zeebe.exporter.spi.Exporter;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.protocol.intent.IncidentIntent;
+import io.zeebe.protocol.intent.WorkflowInstanceIntent;
+import io.zeebe.test.ClientRule;
+import io.zeebe.test.EmbeddedBrokerRule;
+import io.zeebe.test.util.TestUtil;
+import io.zeebe.test.util.record.RecordingExporter;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import org.junit.rules.ExternalResource;
+
+/**
+ * JUnit test rule to facilitate running integration tests for exporters.
+ *
+ * <p>Sets up an embedded broker, gateway, client, and provides convenience methods to:
+ *
+ * <ol>
+ *   <li>run a sample workload (e.g. deploy & create workflows, setup a job worker, etc.)
+ *   <li>visit all exported records so far
+ *   <li>perform simple operations such as deployments, starting a job worker, etc.
+ * </ol>
+ *
+ * An example integration test suite could look like:
+ *
+ * <pre>
+ * public class MyExporterIT {
+ *   @Rule public final ExporterIntegrationRule exporterIntegrationRule = new ExporterIntegrationRule();
+ *
+ *   @Test
+ *   public void shouldExportRecords() {
+ *     // given
+ *     final MyExporterConfig config = new MyExporterConfig();
+ *     exporterIntegrationRule.configure("myExporter", MyExporter.class, config);
+ *     exporterIntegrationRule.startBroker();
+ *
+ *     // when
+ *     exporterIntegrationRule.performSampleWorkload();
+ *
+ *     // then
+ *     exporterIntegrationRule.visitExportedRecords(r -> {
+ *       // code to assert the record was exported to the external system
+ *     });
+ *   }
+ * }
+ * </pre>
+ *
+ * NOTE: calls to the various configure methods are additive, so it is possible to configure more
+ * than one exporter, as long as the IDs are different.
+ */
+public class ExporterIntegrationRule extends ExternalResource {
+
+  public static final BpmnModelInstance SAMPLE_WORKFLOW =
+      Bpmn.createExecutableProcess("testProcess")
+          .startEvent()
+          .intermediateCatchEvent(
+              "message", e -> e.message(m -> m.name("catch").zeebeCorrelationKey("$.orderId")))
+          .serviceTask("task", t -> t.zeebeTaskType("work").zeebeTaskHeader("foo", "bar"))
+          .endEvent()
+          .done();
+
+  private final EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule(false);
+  private ClientRule clientRule;
+  private boolean configured;
```

Comment:
Remove as we should check the broker cfg instead also allow to configure your exporter using the default `zeebe.test.cfg.toml`.